lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -727,6 +727,24 @@ private function gatherRelationJoinColumns(
                     $columnOptions['precision'] = $fieldMapping['precision'];
                 }
 
+                $theJoinTableCharset = $theJoinTable->hasOption('charset') ? $theJoinTable->getOption('charset') : null;
+                if (
+                    ! isset($columnOptions['customSchemaOptions']['charset'])
+                    && isset($class->table['options']['charset'])
+                    && $theJoinTableCharset !== $class->table['options']['charset']
+                ) {
+                    $columnOptions['customSchemaOptions']['charset'] = $class->table['options']['charset'];
+                }
+
+                $theJoinTableCollation = $theJoinTable->hasOption('collation') ? $theJoinTable->getOption('collation') : null;
+                if (
+                    ! isset($columnOptions['customSchemaOptions']['collation'])
+                    && isset($class->table['options']['collate'])
```

Comment:
:thinking: why is it sometimes `collation`, sometimes `collate`? There is little to no mention of `collate` in the ORM, introducing it here seems weird…

Reply:
This is what I found about this:

For columns, it should be `collation`, see https://github.com/doctrine/dbal/blob/719663b15983278227669c8595151586a2ff3327/src/Platforms/AbstractPlatform.php#L2524-L2525 

For tables, `collate` is converted to `collation` and then used, see https://github.com/doctrine/dbal/blob/719663b15983278227669c8595151586a2ff3327/src/Platforms/AbstractMySQLPlatform.php#L455-L464

I think the `collate` conversion on the table is just legacy, and should not be used. 

So I updated the PR to only use `collation` everywhere. Hope it makes sense now.

Reply:
Ah yes, I'm the author of this last code block you pointed to: https://github.com/doctrine/dbal/commit/a18ffc1a1fde506b17c34e9523fce95f16838359

`collate` will be deprecated in the next minor, so this is the way to go.